ext/node/resolution.rs

```diff
@@ -339,7 +339,7 @@ fn resolve_package_target_string(
     ));
   }
   let invalid_segment_re =
-    Regex::new(r"(^|\|/)(..?|node_modules)(\|/|$)").expect("bad regex");
+    Regex::new(r"(^|\\|/)(\.\.?|node_modules)?(\\|/|$)").expect("bad regex");
```

Comment:
The corresponding regex in the node codebase:

https://github.com/nodejs/node/blob/f5dc92cff34933fc4a4685765afbba085f081b36/lib/internal/modules/esm/resolve.js#L299

After cleanup I found that not only escaping symbols are missed but also the question mark

Reply:
I reverted the optional regex group change because I found that it was introduced in [this PR (related to double slash deprecation)](https://github.com/nodejs/node/pull/44477), after [the deno impelementation](https://github.com/denoland/deno/pull/15484) so I guess, it's enough to have the deprecated version for now